go/arrow/array/binarybuilder.go

```diff
@@ -319,7 +320,10 @@ func (b *BinaryBuilder) UnmarshalOne(dec *json.Decoder) error {
 	case string:
 		data, err := base64.StdEncoding.DecodeString(v)
 		if err != nil {
-			return err
+			data, err = hex.DecodeString(v)
+			if err != nil {
+				return err
+			}
```

Comment:
this seems odd to add here. Should we also add this to `AppendValueFromString` and update the docs that you can also use hex to specify binary strings?

Reply:
Hmm....Prob it was in another branch and ended up here. will remove